.travis.yml

```diff
@@ -28,17 +28,17 @@ addons:
       - llvm-toolchain-precise
 
 env:
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Ninja"
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
-
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=gcc CMAKE_GENERATOR="Ninja"
-  - COMPILER=gcc CMAKE_GENERATOR="Unix Makefiles"
+  # Test all generators and Clang + GCC
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=gcc         CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=gcc         CMAKE_GENERATOR="Unix Makefiles"
+
+  # Benchmarks
+  - BENCHMARK=true COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=gcc         CMAKE_GENERATOR="Unix Makefiles"
```

Comment:
> Steps 2-4 are very quick to perform, so there is only a very small possibility that two jobs get in conflict. And even if that happens, we actually try the 2-4 steps three times before giving up and failing. So this can fail, but we'd have to be very unlucky. It has not failed yet for Hana, anyway.

I see, in any case, Isn't there a way to tell git to "push merging"?

> I don't think it adds too much complexity. Basically,
> 
>    1. We must support all the relevant compilers in the benchmarks.
>    2. We must be able to pick which compiler we're displaying on the gh-pages site.

I'm actually concerned about (2), but ok, let's take a step at a time. I'd propose the following roadmap:

1. get gh-pages to display time benchmarks
2. add other aspects
3. add multiple compilers

Meanwhile, benchmarking on travis for several compilers doesn't hurt of course. Till we make it to (3) we just choose one of the compiler versions to actually push results to gh-pages.